src/askem_beaker/contexts/mira_model_edit/procedures/python3/model_preview.py

```diff
@@ -1,10 +1,20 @@
 from IPython.core.interactiveshell import InteractiveShell;
 from IPython.core import display_functions;
 from mira.modeling.amr.petrinet import template_model_to_petrinet_json
+from mira.modeling.amr.stockflow import template_model_to_stockflow_json;
+from mira.modeling.amr.regnet import template_model_to_regnet_json;
 
 format_dict, md_dict = InteractiveShell.instance().display_formatter.format(GraphicalModel.for_jupyter(model))
+
+if "{{ schema_name }}" == "regnet":
+    model_json = template_model_to_regnet_json({{ var_name|default("model") }})
+elif "{{ schema_name }}" == "stockflow":
+    model_json = template_model_to_stockflow_json({{ var_name|default("model") }})
+else:
+    model_json = template_model_to_petrinet_json({{ var_name|default("model") }})
+
```

Comment:
Not sure how universal this would be, but this might be able to be shortened as follows using the templating:

```suggestion

model_json = template_model_to_{{ schema_name|default("petrinet") }}_json({{ var_name|default("model") }})

```

This is, of course, assuming that there will always be a `template_mode_to_SCHEMA_json` for all schemas we are handling.

Reply:
will leave things alone for now